src/Symfony/Component/Config/Resource/ClassExistenceResource.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Resource;
+
+/**
+ * ClassExistenceResource represents a class existence.
+ * Freshness is only evaluated against resource existence.
+ *
+ * The resource must be a fully-qualified class name.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class ClassExistenceResource implements SelfCheckingResourceInterface, \Serializable
+{
+    private $resource;
+    private $exists;
+
+    /**
+     * @param string $resource The fully-qualified class name
+     */
+    public function __construct($resource)
+    {
+        $this->resource = $resource;
+        $this->exists = class_exists($resource);
```

Comment:
we should use ``class_exists() || interface_exists() || trait_exists()`` to make it usable for interfaces and traits too